src/System.Net.NetworkInformation/src/System/Net/NetworkInformation/LinuxIPv4InterfaceProperties.cs

```diff
@@ -0,0 +1,78 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.IO;
+
+namespace System.Net.NetworkInformation
+{
+    internal class LinuxIPv4InterfaceProperties : IPv4InterfaceProperties
+    {
+        private LinuxNetworkInterface _linuxNetworkInterface;
+
+        public LinuxIPv4InterfaceProperties(LinuxNetworkInterface linuxNetworkInterface)
+        {
+            _linuxNetworkInterface = linuxNetworkInterface;
+        }
+
+        public override int Index
+        {
+            get
+            {
+                return _linuxNetworkInterface.Index;
+            }
+        }
+
+        public override bool IsAutomaticPrivateAddressingActive
+        {
+            get
+            {
+                throw new PlatformNotSupportedException();
+            }
+        }
+
+        public override bool IsAutomaticPrivateAddressingEnabled
+        {
+            get
+            {
+                throw new PlatformNotSupportedException();
+            }
+        }
+
+        public override bool IsDhcpEnabled
+        {
+            get
+            {
+                throw new PlatformNotSupportedException();
+            }
+        }
+
+        public override bool IsForwardingEnabled
+        {
+            get
+            {
+                // /proc/sys/net/ipv4/conf/<name>/forwarding
+                string path = Path.Combine(LinuxNetworkFiles.Ipv4ConfigFolder, _linuxNetworkInterface.Name, LinuxNetworkFiles.ForwardingFileName);
+                return int.Parse(File.ReadAllText(path)) == 1;
+            }
+        }
+
+        public override int Mtu
+        {
+            get
+            {
+                // /proc/sys/net/ipv4/conf/<name>/mtu
+                string path = Path.Combine(LinuxNetworkFiles.Ipv4ConfigFolder, _linuxNetworkInterface.Name, LinuxNetworkFiles.MtuFileName);
+                return int.Parse(File.ReadAllText(path));
```

Comment:
If we expect properties like this to be accessed repeatedly, we should spend some time optimizing them via caching a bunch of stuff.